### PR TITLE
Adding Google login [WIP]

### PR DESCRIPTION
**Pull request by Jolanrensen**
I'm currently on vacation so I cannot work on it, but I am willing to contribute my changes to the project, so I'm creating a pull request in advance.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -30,6 +30,8 @@ dependencies {
     implementation 'androidx.core:core-ktx:1.0.2'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
     implementation 'com.google.android.material:material:1.0.0'
+    implementation "com.google.android.gms:play-services-auth:17.0.0"
```
Comment (samtstern) [inline app/build.gradle:33]: If we add this dependency we are no longer showing auth without play services ... we're just showing Auth without the Firebase SDK.

I understand this is useful for Android Wear (which has Google Sign In but not Firebase Auth) but the main use cases for this repo are for devices without Play services entirely like Kindle Fire or Huawei

Comment (Jolanrensen) [reply app/build.gradle:33]: Yup, you're right. We can indeed keep it separate on my branch for people who want to use it for Wear OS while not interfering with devices without Google Sign-In. I'll rebase my branch on the master to update it and then we can maybe create a link to my fork in the readme as suggested in the email.

**Comment by Jolanrensen** [comment]
Alright, it can be tested!
I already tested it partially for my own app, but testing by other is much appreciated of course.

**Comment by samtstern** [comment]
@Jolanrensen thanks for rebasing!  Your branch looks good.  So should we close this PR and instead just add a new section to the `README` here linking to your fork for those who want to learn about using this technique for Google Sign In?

**Comment by ElectricMagic** [comment]
We have a single apk running on Amazon devices, on https://konstakang.com/devices/rpi4/ and on regular GooglePlayServices-supporting devices. For the latter we can use firebaseauth directly, but for the others we use auth-without-play-services.

The trick to getting things to coexist is to have in `AndroidManifest.xml` a clause like:
```
<service android:name="com.google.firebase.components.ComponentDiscoveryService">
	<!--Remove the firebase standard registrar-->
	<meta-data
		android:name="com.google.firebase.components:com.google.firebase.auth.FirebaseAuthRegistrar"
		tools:node="remove"/>
	<!--Add ours-->
	<meta-data
		android:name="com.google.firebase.components:com.example.exampleapp.ProxyFirebaseAuthRegistrar"
		android:value="com.google.firebase.components.ComponentRegistrar" />
</service>
```
The implementation of `ProxyFirebaseAuthRegistrar.getComponents` calls `GoogleApiAvailability.isGooglePlayServicesAvailable`, and delegates to FirebaseAuth or FirebaseRestAuth as appropriate.

A revision of this mechanism might allow this third scenario you're exploring, where you do have google signin, but don't have firebaseauth, and to do so in a single branch.

**Comment by samtstern** [comment]
Wow @ElectricMagic that's a very cool (and advanced) hack!  Thanks for sharing.

**Comment by ElectricMagic** [comment]
Thank you @samstern!

While I have your ear, do you expect `com.firebase.ui.auth` to be usable on Android TV? I'm finding that it doesn't handle EditText widgets at all correctly. Even the FirebaseUI-Android demo app won't show the leanback keyboard in `CheckEmailFragment` (because the `OnClickListener` absorbs the 'click'), and none of the screens seem to have proper handling of focus with a leanback keyboard around.

Is it just that it's expected that signin on Android TV will be using the google signin that basically has to be there? For our app we're going to need to handle some other auth providers, and will need working text entry on some of them at least.

**Comment by samtstern** [comment]
@ElectricMagic no I wouldn't expect FirebaseUI to work on Android TV, nobody has ever tested it there (afaik).  If you have simple fixes you think would improve it feel free to send us PRs!  But phones are our only priority.

Also for future FirebaseUI discussions let's use that repo so we don't bug @Jolanrensen

**Comment by samtstern** [comment]
@Jolanrensen @ElectricMagic I am sorry we never got this all the way merged in but I have some good news ... as of version `20.0.0` of the Firebase Auth SDK we no longer require Google Play services!
https://firebase.google.com/support/release-notes/android#auth_v20-0-0

This means this entire repository is no longer necessary.

**Pull request closed without merging**